### PR TITLE
Changed casting for write offset. 

### PR DESCRIPTION
**Pull request by bgiorgio0506**
I have changed the casting for type Int64  from Int32Value to Int64Value. This casting was creating an issue while writing on Int64 offset types.

I have tested the code as part of the process of validating changes for another porject. 
Also to note the Typescript definitions are wrong for types Int64 and Uint64 as the function requires 4 arguments for those types and not 3 ( I will open a new PR for TS types).

### CODE REVIEW

**Review by koesie10 — COMMENTED** [review]
Thanks! Could you also make the equivalent change on line 343 for uint64?

**Comment by bgiorgio0506** [comment]
> Thanks! Could you also make the equivalent change on line 343 for uint64?

yes sure I will do it in the coming weeks ; )

**Comment by bgiorgio0506** [comment]
@koesie10 you can merge if my changes satisfy your code standards

File: packages/fsuipc/src/FSUIPC.cc [hunk 1]
```diff
@@ -340,7 +340,7 @@ void FSUIPC::Write(const Napi::CallbackInfo& info) {
             std::string(info[3].As<Napi::String>().Utf8Value().c_str());
         x = std::stoll(x_str);
       } else if (info[3].IsNumber()) {
-        x = (uint64_t)info[3].ToNumber().Uint32Value();
+        x = info[3].As<Napi::BigInt>().Uint64Value(false);
```
Comment (koesie10) [inline packages/fsuipc/src/FSUIPC.cc:343]: Why do we need to create a bigint before we're able to get the uint64 value? I would expect this to work the same as the `Int64Value`.

File: package-lock.json
Comment (koesie10) [inline package-lock.json:1]: Can you please remove this file and revert the changes to `yarn.lock`?

**Review by koesie10 — COMMENTED** [review]
Thanks again for your contribution and sorry for my late response. I still have a question and a suggestion before this can be merged.

File: packages/fsuipc/src/FSUIPC.cc (hunk 1 above)
Comment (bgiorgio0506) [reply packages/fsuipc/src/FSUIPC.cc:343]: @koesie10  this is bc uint64 is not a member of Number you can cast it only through BigInt  see Napi doc https://github.com/nodejs/node-addon-api/blob/main/doc/bigint.md and https://github.com/nodejs/node-addon-api/blob/main/doc/number.md

File: package-lock.json
Comment (bgiorgio0506) [reply package-lock.json:1]: yes sure will do

**Comment by bgiorgio0506** [comment]
@koesie10 sorry for the delay I have reverted the yarn.lock file and removed the package-lock.json. I have answerd your previous question let me know if you need some more clarification

**Comment by koesie10** [comment]
@bgiorgio0506 Thanks! I've cherry-picked your changes in #33 and made some additional changes such that the types are now correct.

**Comment by koesie10** [comment]
@bgiorgio0506 This has been published in version v0.7.2.

**Comment by bgiorgio0506** [comment]
@koesie10  thanks !
